### PR TITLE
Fix for duplicate field names and render validation messages for server

### PR DESCRIPTION
**Pull request by jeffibm**
Before -

https://github.com/ManageIQ/ui-components/assets/87487049/f9b88090-1e73-45d6-94b2-b31f8a2a7ad8

image
image
image

- Edit a service dialog, 
- Click edit field to open the options modal box
- Change a field name to an already used name (duplicate name)
- Click save at the options modal box
- Click save at the service dialog form
- We will get a warning that the field name cannot be used.
- cancel the form.
- select the service dialog again and edit it
- check the field name of the field where you tried to put a duplicate name.
- See that the field name got updated and now we have a duplicate field.

After

https://github.com/ManageIQ/ui-components/assets/87487049/f3b8c739-6b52-45b3-92f2-8278d54fc7d2

image

- The options modal box will check for duplicate field names
- If there are duplicate field names, the save button of the options modal will be disabled.
- Introduced Validation messages in the options modal box. Earlier it was displayed as a title (on mouse hover) of the save button
- For existing records, the service dialog save button will be disabled if there are duplicate field names.
- The options modal's save button will also be disabled unless you change the field name to avoid duplicate.

### CODE REVIEW

File: src/dialog-editor/services/dialogValidationService.ts
Comment (Fryguy) [inline src/dialog-editor/services/dialogValidationService.ts:112]: I think we should present this more directly.  Something like

```suggestion
      this.invalid.message = `Unable to save due to duplicate field name "${field.name}"`;
```

Comment (Fryguy) [reply src/dialog-editor/services/dialogValidationService.ts:112]: Separate question - is it possible to add i18n support for this line?

**Comment by Fryguy** [comment]
@DavidResende0 Can you also review here?

**Review by DavidResende0 — APPROVED** [review]
LGTM, just look into @Fryguy's comments before merging.

File: src/dialog-editor/components/modal/modalComponent.ts
Comment (DavidResende0) [inline src/dialog-editor/components/modal/modalComponent.ts:163]: This should probably be marked for translation.

**Review by Fryguy — APPROVED** [review]
LGTM - @DavidResende0 if you are good please merge.  Note that we will have to cut a new release once it's merged.

**Pull request merged**
